apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts

```diff
@@ -0,0 +1,61 @@
+import { Agent, AnonymousIdentity, HttpAgent } from "@dfinity/agent"
+import { decodeIcrcAccount } from "@dfinity/ledger-icrc"
+
+import { exchangeRateService } from "@nfid/integration"
+
+import { idlFactory as icrc1and2IDL } from "../../idl/token-pepe-ledger"
+import { RPCMessage } from "../../type"
+import { actorService } from "../actor.service"
+import { IC_HOSTNAME } from "../method/interactive/icrc49-call-canister-method.service"
+import { TransferMetadata } from "./interfaces"
+
+export const getIcrc1TransferMetadata = async (
+  message: MessageEvent<RPCMessage>,
+  args: any,
+): Promise<TransferMetadata> => {
+  const delegation = new AnonymousIdentity()
+
+  const agent: Agent = new HttpAgent({
+    host: IC_HOSTNAME,
+    identity: delegation,
+  })
+
+  const actor = actorService.getActor<any>(
+    message.data.params.canisterId,
+    icrc1and2IDL,
+    agent as never,
+  )
+
+  const { owner, subaccount } = decodeIcrcAccount(message.data.params.sender)
+
+  const [balance, symbol, decimals, fee] = await Promise.all([
+    actor.icrc1_balance_of({
+      owner: owner,
+      subaccount: subaccount ?? [],
+    }),
+    actor.icrc1_symbol(),
+    actor.icrc1_decimals(),
+    actor.icrc1_fee(),
+  ])
+
+  const [requestParams] = JSON.parse(args)
+
+  const amount = Number(requestParams.amount)
+  const isInsufficientBalance = amount + Number(fee) > Number(balance)
```

Comment:
are you sure that number works fine with 16+ decimails?

Reply:
changed to strings everywhere, main component which renders all numbers in app changed to accept number or string and convert to bignumbers under the hood for backward compatibility